src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -143,31 +154,27 @@
    pool :- jruby-schemas/pool-queue-type]
   (.pollFirst pool timeout TimeUnit/MILLISECONDS))
 
+(schema/defn borrowed-poison-pill
+  [pool :- jruby-schemas/pool-queue-type
+   instance :- PoisonPill]
+  (.putFirst pool instance)
+  (throw (IllegalStateException.
```

Comment:
Is this missing an `if` or something? I don't imagine we want to throw an exception all the time